compositor_render/src/transformations/web_renderer/browser.rs

```diff
@@ -176,9 +176,11 @@ impl cef::RenderHandler for RenderHandler {
     }
 
     fn on_paint(&self, _browser: &cef::Browser, buffer: &[u8], _resolution: Resolution) {
-        self.painted_frames_sender
-            .send(buffer.to_vec())
-            .expect("send frame");
+        if !self.painted_frames_sender.is_full() {
+            self.painted_frames_sender
+                .send(buffer.to_vec())
+                .expect("send frame");
+        }
```

Comment:
I agree, technically concept of channel does not work well here, consider using Arc<Mutex<Vec<u8>>> and just storing last frame.

Or if you always want frame to available when reading you could create `Arc<Mutext<Arc<[u8]>>>` or `Arc<Mutext<Bytes>>`  so when you are taking value from "channel' you don't leave empty value. Empty value still needs to be handled of course because at the beginning you don't have anything to fill it with.